webcurator-ui/src/views/target/target-tabs/TargetTabPanelGeneral/modals/TargetAddPermissionModal.vue

```diff
@@ -0,0 +1,242 @@
+import { formatDate } from '@/utils/helper';
+import { type UseFetchApis, useFetch } from '@/utils/rest.api';
+
+import Loading from '@/components/Loading.vue';
+
+const dialogRef: any = inject('dialogRef');
+
+const rest: UseFetchApis = useFetch();
+
+interface HarvestAuth {
+  id: number;
+  name: string;
+  authorisingAgents: any[];
+  permissions: any[];
+}
+
+const returnedHarvestAuths = ref<HarvestAuth[]>([]);
+const loading = ref(false);
+const loadingPermission = ref(false);
+
+const harvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '', urlPatterns: [] }[]>([]);
+const filteredHarvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '',  urlPatterns: []  }[]>([]);
+
+const searchTerm = ref('');
+
+const seed = ref();
+let expandedPermission = reactive<Permission>({} as Permission);
+const expandedRows = ref<any[]>([]);
+const permissionStatuses = ref();
+
+const prepareData = (data: HarvestAuth[]) => {
+  data.forEach((harvestAuth: HarvestAuth) => {
+    if (harvestAuth.authorisingAgents.length > 0) {
+      harvestAuth.authorisingAgents.forEach((authorisingAgent: any) => {
+        if (authorisingAgent.permissions.length > 0) {
+          harvestAuths.value.push({
+            id: harvestAuth.id,
+            name: harvestAuth.name,
+            agent: authorisingAgent.name,
+            permissionId: authorisingAgent.permissions[0].id,
+            startDate: authorisingAgent.permissions[0].startDate,
+            endDate: authorisingAgent.permissions[0].endDate,
+            urlPatterns: authorisingAgent.permissions[0].urlPatterns
+          });
+        }
+      })
+    }
+  });
+  filteredHarvestAuths.value = harvestAuths.value;
+};
+
+const setExpandedRow = async (event: DataTableRowClickEvent) => {
+  loadingPermission.value = true;
+  const isExpanded = (expandedRows.value as any[]).find((p) => p.id === event.data.id)
+
+  if (isExpanded?.id) {
+    expandedRows.value = [event.data] as any
+    const fetchedPermission: any = await usePermissionStore().fetch(event.data.permissionId);
+    
+    expandedPermission = fetchedPermission;
+  } else {
+    expandedRows.value = []; 
+    expandedPermission = ({} as Permission);
+  }
+
+  loadingPermission.value = false;
+}
+
+const fetch = () => {
+  const searchParams = {
+    offset: 0,
+    limit: 1024,
+  };
+
+  loading.value = true;
+
+  rest
+    .post('harvest-authorisations', searchParams, { header: 'X-HTTP-Method-Override', value: 'GET' })
+    .then((data: any) => {
+      returnedHarvestAuths.value = data['harvestAuthorisations'];
+      prepareData(returnedHarvestAuths.value);
+      loading.value = false;
+    })
+    .catch((err: any) => {
+      console.log(err.message);
+      loading.value = false;
+    });
+};
+
+const search = () => {
+  const lowerCaseSearchTerm = searchTerm.value.toLowerCase();
+  filteredHarvestAuths.value = harvestAuths.value.filter((g: any) => 
```

Comment:
The rest api permission search (Permissions.java -> get()) is ready now, so the search method can be implemented base on the search api. With the rest api can match the permissions by seed url.